presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/PlanNodeMatcher.java

```diff
@@ -33,8 +33,11 @@ public PlanNodeMatcher(Lookup lookup)
     }
 
     @Override
-    public <T> Stream<Match<T>> matchWith(WithPattern<T> withPattern, Object object, Captures captures)
+    public <T, C> Stream<Match<T>> matchWith(WithPattern<T> withPattern, Object object, Captures captures, C context)
```

Comment:
Can you explain why and also put that in the commit message.

Reply:
The idea is to pass `Rule.Context` or just `Lookup` as context parameter for pattern matching. So user of pattern matching who provides `Predicates` or property `Functions` to `Patterns` will be able to access `Lookup` in them.